src/internal.c

```diff
@@ -687,7 +687,7 @@ int GenerateKey(byte hashId, byte keyId,
         return WS_BAD_ARGUMENT;
     }
 
-    digestSz = wc_HashGetDigestSize(hashId);
+    digestSz = wc_HashGetDigestSize((enum wc_HashType)hashId);
```

Comment:
Casting hashId to (enum wc_HashType) so many times seems a little messy.  Could we make a new variable and do the cast once?  Or, alternatively change the prototype for GenerateKey() to accept a "enum wc_HashType" for hashId?  GenerateKey() is an internal/local function, so changing an argument type should be ok.

Reply:
Thank you for the review. Introduced new variable instead of casting each call.